Userland/Applications/Screenshoter/ScreenshoterWidget.cpp

```diff
@@ -0,0 +1,181 @@
+#include <LibGUI/WindowServerConnection.h>
+#include <LibGUI/Button.h>
+#include <LibGUI/RadioButton.h>
+#include <LibGUI/SpinBox.h>
+#include <LibGUI/CheckBox.h>
+#include <LibGUI/Label.h>
+#include <LibGUI/Notification.h>
+#include <LibGfx/PNGWriter.h>
+#include <LibGfx/Palette.h>
+#include <serenity.h>
+
+class SelectableLayover final : public GUI::Widget {
+    C_OBJECT(SelectableLayover)
+public:
+    SelectableLayover(GUI::Window* window)
+        : m_window(window)
+        , m_background_color(palette().threed_highlight().with_alpha(128))
+    {
+        set_override_cursor(Gfx::StandardCursor::Crosshair);
+    }
+
+    virtual ~SelectableLayover() override {};
+
+    Gfx::IntRect region() const
+    {
+        return m_region;
+    }
+
+private:
+    virtual void mousedown_event(GUI::MouseEvent& event) override
+    {
+        if (event.button() == GUI::MouseButton::Left)
+            m_anchor_point = event.position();
+    };
+
+    virtual void mousemove_event(GUI::MouseEvent& event) override
+    {
+        if (m_anchor_point.has_value()) {
+            m_region = Gfx::IntRect::from_two_points(*m_anchor_point, event.position());
+            update();
+        }
+    };
+
+    virtual void mouseup_event(GUI::MouseEvent& event) override
+    {
+        if (event.button() == GUI::MouseButton::Left)
+            m_window->close();
+    };
+
+    virtual void paint_event(GUI::PaintEvent&) override
+    {
+        GUI::Painter painter(*this);
+        painter.clear_rect(m_window->rect(), Gfx::Color::Transparent);
+        painter.fill_rect(m_window->rect(), m_background_color);
+
+        if (m_region.is_empty())
+            return;
+
+        painter.clear_rect(m_region, Gfx::Color::Transparent);
+    }
+
+    virtual void keydown_event(GUI::KeyEvent& event) override
+    {
+        if (event.key() == Key_Escape) {
+            m_region = Gfx::IntRect();
+            m_window->close();
+        }
+    }
+
+    Optional<Gfx::IntPoint> m_anchor_point;
+    Gfx::IntRect m_region;
+    GUI::Window* m_window = nullptr;
+    Gfx::Color const m_background_color;
+};
+
+ScreenshoterWidget::ScreenshoterWidget()
+{
+    load_from_gml(screenshoter_window_gml);
+
+    m_whole_button = *find_descendant_of_type_named<GUI::RadioButton>("wholescreen");
+    m_active_button = *find_descendant_of_type_named<GUI::RadioButton>("activewindow");
+    m_custom_button = *find_descendant_of_type_named<GUI::RadioButton>("customregion");
+    m_delay_spinbox = *find_descendant_of_type_named<GUI::SpinBox>("delay");
+    m_copy_checkbox = *find_descendant_of_type_named<GUI::CheckBox>("copy_to_clipboard");
+    m_cancel_button = *find_descendant_of_type_named<GUI::Button>("cancel_button");
+    m_ok_button = *find_descendant_of_type_named<GUI::Button>("ok_button");
+
+    m_cancel_button->on_click = [&](auto) {
+        GUI::Application::the()->quit();
+    };
+
+    m_ok_button->on_click = [&](auto) {
+        GUI::Application::the()->active_window()->close();
```

Comment:
We don't need to ask `Application` what the active window is, it's the window we are in:
```cpp
window()->close();
```